### PR TITLE
Express server for testing

### PR DESCRIPTION
**Pull request by yablochko8**
| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit b0ce8ca26f8f7ca1bbc156faf295c3764e237d82  | 
|--------|--------|

### Summary:
Introduced an Express server for testing, added new endpoints, restructured the project directory, and updated services and configurations to support the new functionality.

**Key points**:
- Added `server/expressServer.ts` to create an Express server with endpoints `/express/heartbeat`, `/express/recent-commits`, and `/express/full-flow`.
- Updated `frontend/src/App.tsx` to interact with the new Express server endpoints.
- Moved and renamed several configuration files to the `database` and `frontend` directories.
- Updated `server/.env.example` with new environment variables for local testing.
- Added `server/services/github/getCommits.ts` and `server/services/github/getRepos.ts` for GitHub API interactions.
- Added `server/services/openai/evaluateCommits.ts` for OpenAI API interactions.
- Updated `server/services/reviewer.ts` to use new GitHub and OpenAI services.
- Added `server/services/slack/postText.ts` for posting messages to Slack.
- Removed obsolete files and code, including `server/server.ts` and `server/services/callGithub.ts`.
- Updated `server/package.json` with new dependencies and scripts.
- Updated `README.md` with new project structure information.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import "./App.css";
+
+export const PORT = 4101; // change this to an import before doing anything serious
+
+const serverPath = `http://localhost:${PORT}`;
```
Comment (dcsan) [inline frontend/src/App.tsx:6]: could move the server path into a .env or config file, so we can run this when deployed
its a bit of a pain to set this all up with vite
also depends on the deployed env as sometimes you can just inject variables into the app on the deployed dashboard (eg using vercel or cloudflare)

**Review by dcsan — APPROVED** [review]
lots of changes here, i didn't review in detail 
if the website is mainly to test the APIs, you could also consider just unit tests or something to run from server. 

there are also vscode extensions to just load an http endpoint
https://marketplace.visualstudio.com/items?itemName=humao.rest-client

i find this easier than setting up a whole separate postman project.
you can also then document the endpoints with a `.http` file in the repo.

File: server/expressServer.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    // const result = recentCommits.map((commit) => ({
+    //   repo: commit.repo,
+    //   user: commit.user,
+    //   message: commit.message,
+    //   date: commit.time,
+    // }));
+    res.json({ commits: recentCommits });
+  } catch (error) {
+    console.error("Error requesting reviews:", error);
+    res.status(500).json({ error: "Failed to fetch reviews." });
```
[the remark this answers is not in the document]
Comment (yablochko8) [reply server/expressServer.ts:33]: thanks bot

File: frontend/src/App.tsx (hunk 1 above)
Comment (yablochko8) [reply frontend/src/App.tsx:6]: good shout

**Pull request merged**